2.0 datasetId

HarryDunham opened the pull request:
* switch to `filename` as a field/param to specify
* Switch to NMT request body leveraging the datasetID rather than a dict
* Update SaveDatasetResponse for new fields

Commit:
update save dataset

Commit:
switch datasets to datasetId

Commit:
update tests for request changes

bradnussbaum reviewed (changes requested):
We need to discuss this more.

Commit:
update appropriate response fields

Commit:
switch to key + datasetId

Commit:
update nmt train pipeline with datasetId

HarryDunham commented:
I've updated this PR and rebased it on top of the latest 2.0. I've also added changes for the NMT training pipeline since that was leveraging `datasets` rather than `datasetId`
